oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentStore.java

```diff
@@ -486,4 +490,45 @@ default int getNodeNameLimit() {
     default Throttler throttler() {
         return NO_THROTTLING;
     }
+
+    /**
+     * Get a list of documents with only projected fields (as mentioned in projections param)
+     * along with "_id" field and where the key is greater than a start value and
+     * less than an end value <em>and</em> the given "indexed property" is greater
+     * or equals the specified value.
+     * <p>
+     * The indexed property can either be a {@link Long} value, in which case numeric
+     * comparison applies, or a {@link Boolean} value, in which case "false" is mapped
+     * to "0" and "true" is mapped to "1".
+     * <p>
+     * The returned documents are sorted by key and are immutable.
+     *
+     * @param <T> the document type
+     * @param collection the collection
+     * @param fromKey the start value (excluding)
+     * @param toKey the end value (excluding)
+     * @param indexedProperty the name of the indexed property (optional)
+     * @param startValue the minimum value of the indexed property
+     * @param limit the maximum number of entries to return
+     * @param projections {@link List} of projected keys (optional). Keep this empty to fetch all fields on document.
```

Comment:
This contract `to fetch all if projections are empty` is in line with how mongo behaves.
If we provide an empty projection array, then mongo returns all the results.
In case we want specific fields, then we need to mention those individually.

Reply:
"This is what MongoDB does" IMHO is not sufficient reason for something that would apply to all DocumentStore implementations. Leaking implementation details...